### ISSUE TITLE
Upgrade Navigator with Octree Controls

### ISSUE THREAD

**Issue by Birdasaur** [trinity-xai/Trinity#85]
Upgrade the existing Content Navigator (NavigatorPane) to allow user to traverse points in 3D projection space using "nearest neighbor" steps. This will require leveraging the Octree support that was added to the Shape3DControlPane but never really integrated as part of any useful workflow.

**Comment by Birdasaur** [comment]
@samypr100 For this upcoming pull request, should we be updating the Copyright comments at the source headers?
The copyright years that is?

**Comment by samypr100** [comment]
The headers are not needed in new code, only when disambiguation is needed from the project's overall copyright

**Comment by Birdasaur** [comment]
Do you mean disambiguation is needed for source files which existed prior to when we open sourced to github?

**Comment by samypr100** [comment]
> Do you mean disambiguation is needed for source files which existed prior to when we open sourced to github?

Yes, but that's already done

**Comment by Birdasaur** [comment]
What about pre-existing files that had to be changed recently? Is it acceptable to add the new (current ) copyright line to the source file below the original?

And if so... what is the most appropriate way?
`
/* Copyright (C) 2021 - 2023 The Johns Hopkins University Applied Physics Laboratory LLC */
/* Copyright (C) 2021 - 2025 Sean Phillips */
`
or

`
/* Copyright (C) 2021 - 2023 The Johns Hopkins University Applied Physics Laboratory LLC */
/* Copyright (C) 2024 - 2025 Sean Phillips */
`

Also why can't we add your name to the copyright???

**Comment by samypr100** [comment]
All the files were fixed in https://github.com/trinity-xai/Trinity/pull/81 we shouldn't need to modify them anymore, even the copyright year shouldn't be need to be updated

**Comment by Birdasaur** [comment]
oh...
Ok then I'm dumb. I thought I was supposed to update copyright years whenever a file was updated.
Is that bad? Should I go back through and fix all my changed files?

**Comment by samypr100** [comment]
Luckily, copyright year does not need to be updated

**Comment by Birdasaur** [comment]
Ok... let me clean up these changes a bit.

**Comment by Birdasaur** [comment]
oh wait, can you clarify something...
We put the JHUAPL copyright on original files for the years 2021 - 2023 to indicate that they were the organization that produced the source during that time right? (and therefore they have copyright for any changes through that period)

So that means any and all subsequent files and changes to existing files are automatically under our copyright?
In otherwords how does someone look at a source file and distinguish between who and when copyright is applied?

**Comment by samypr100** [comment]
> > oh wait, can you clarify something... We put the JHUAPL copyright on original files for the years 2021 - 2023 to indicate that they were the organization that produced the source during that time right? (and therefore they have copyright for any changes through that period)
> > So that means any and all subsequent files and changes to existing files are automatically under our copyright? In otherwords how does someone look at a source file and distinguish between who and when copyright is applied?
> 
> Right now, bsed on the copyright md which states changes from which commit are copyrighted differently

If I messed up the years, feel free to fix

**Comment by Birdasaur** [comment]
ahhhhh ok I get it now. So absolutely no need to insert additional copyright statements into new files or changed files
right?

**Comment by samypr100** [comment]
> ahhhhh ok I get it now. So absolutely no need to insert additional copyright statements into new files or changed files right?

Right, unless you need to disambiguate from https://github.com/trinity-xai/Trinity/blob/main/COPYRIGHT.md (e.g. you used a file from some other open source project)
